project.clj

```diff
@@ -141,7 +141,7 @@
                                   (:out (clojure.java.shell/sh
                                          "git" "symbolic-ref" "--short" "HEAD")))})}]
 
-  :global-vars {*warn-on-reflection* true}
```

Comment:
Moved to the `:ci` profile because it also affects 3rd party libs, which possibly results in a wall of warns (depending on what you have in your ~/.lein) on repl startup:

<img width="1105" alt="Screenshot 2020-12-27 at 11 33 42" src="https://user-images.githubusercontent.com/1162994/103169451-10154880-483c-11eb-9ad1-786d424e56c4.png">

IROH's approach is better, since it affects only the project itself and also actively fails the build.

We could implement it here now (wasn't possible at all before the summer refactorings I think; code now is more reloadable), as a subsequent PR